pandas/core/internals/managers.py

```diff
@@ -164,15 +164,15 @@ def __nonzero__(self):
     __bool__ = __nonzero__
 
     @property
-    def shape(self):
+    def shape(self) -> Tuple[int, ...]:
         return tuple(len(ax) for ax in self.axes)
 
     @property
     def ndim(self) -> int:
         return len(self.axes)
 
-    def set_axis(self, axis, new_labels):
-        new_labels = ensure_index(new_labels)
+    def set_axis(self, axis: int, new_labels: Index):
+        # Caller is responsible for ensuring we have an Index object.
```

Comment:
if this isn't public, the comment isn't needed with type annotations

Reply:
is mypy smart enough to enforce this?  I'm hoping to ween internals off of `ensure_index` completely, the comment is to keep track of where it has been done

Reply:
my only concern is keeping comments in-sync with code changes over time.

Reply:
> is mypy smart enough to enforce this?

as we add more type hints throughout the code. (the pace should be quicker but too much bikeshedding IMO) if mypy is green, job done! you can only get so far if the type hints are wrong.

Reply:
> the pace should be quicker but too much bikeshedding IMO

Fair enough, im happy to defer to you on these

Reply:
> > the pace should be quicker but too much bikeshedding IMO
> 
> Fair enough, im happy to defer to you on these

The point is, mypy is the checker, so the reviewer's don't need to be. (there does need to be some consideration of style, readability, maintenance etc)

Reply:
> (there does need to be some consideration of style, readability, maintenance etc)

and we could start to address those concerns with tools like https://pypi.org/project/flake8-annotations-complexity/